src/audio/module_adapter/module_adapter.c

```diff
@@ -320,6 +191,19 @@ int module_adapter_prepare(struct comp_dev *dev)
 		return ret;
 	}
 
+#if CONFIG_ZEPHYR_DP_SCHEDULER
+	/* set the period for the DP module unless it has already been calculated by the
+	 * module itself during prepare
+	 * It may happen i.e. for modules like phrase detect that do not produce audio data
+	 * but events and therefore don't have any deadline for processing
+	 * Second example is a module with variable data rate on output (like MPEG encoder)
+	 */
+	if ((mod->dev->ipc_config.proc_domain == COMP_PROCESSING_DOMAIN_DP) && (!dev->period)) {
```

Comment:
all internal parentheses superfluous. It's helpful sometimes to try to imagine how an expression can be misinterpreted if parentheses aren't there. In your case for the second pair how can `if (A && !B->C)` be wrongly ordered?